### PR TITLE
Train resnet model on last 12 layers.

### PR DESCRIPTION
**Pull request by matthew-sochor**
This is the last training unit in the resnet model.
Also, callbacks and ability to re-train from previous best model were added.

This looks like a promising way forward, using our existing data and *not* adding any hidden dense layers I was able to get to:

484s - loss: 0.0083 - categorical_accuracy: 0.9983 - val_loss: 0.8114 - val_categorical_accuracy: 0.8740

So I think we can tweak this a bit if need be but really we should do the "get moar data!" suggestion. 

This is after slowing down the learning rate twice by a factor of 4 (for some reason I swear thats what you suggested to me josiah, but I could be making that up) and picking up from the previous run's best checkpoint.

Oh and the model is slow to train now :(  but if you are patient and doing family stuff it isn't too bad :)

### CODE REVIEW

**Comment by thenomemac** [comment]
Ya slowing the learning rate is super important. It's called learning rate
annealing in the literature

**Comment by thenomemac** [comment]
As Matt Graziano suggested I think we should really focus on adding
features to the app so that the judges can see our vision of what it will
look like and do. That may be more important than the model since I don't
think they really understand how that works anyways right now. I still they
would really like to do the live demo at the next stage and use Remote
Desktop so it actually works. Think we should just go through the dataset
and find the fish with the lowest deviance scores and use those for the
live demo because they will work well

**Comment by thenomemac** [comment]
Didn't mean to downplay this 87% is awesome once we get more data this'll kick butt.

can you also upload the dataset you used. don't know if i have a link to new filtered data?

**Comment by matthew-sochor** [comment]
no worries :) 

dataset is Matt G's latest.  I think I want to call the "model building" part of this done and just fit the best we can after we add in some of our own pictures that we have been taking
